pint_xarray/accessors.py

```diff
@@ -634,7 +637,45 @@ def to(self, units=None, **unit_kwargs):
         Data variables:
             a        (x) float64 <Quantity([   0.  250.  500.  750. 1000.], 'millimet...
             b        (x) float64 <Quantity([-1000.  -750.  -500.  -250.     0.], 'gra...
+
+        Convert homogeneous data
+
+        >>> ds = xr.Dataset(
+        ...     data_vars={
+        ...         "a": ("x", np.linspace(0, 1, 5) * ureg.kg),
+        ...         "b": ("x", np.linspace(-1, 0, 5) * ureg.mg),
+        ...     },
+        ...     coords={"u": ("x", np.arange(5) * ureg.s)},
+        ... )
+        >>> ds
+        <xarray.Dataset>
+        Dimensions:  (x: 5)
+        Coordinates:
+            u        (x) int64 [s] 0 1 2 3 4
+        Dimensions without coordinates: x
+        Data variables:
+            a        (x) float64 [kg] 0.0 0.25 0.5 0.75 1.0
+            b        (x) float64 [mg] -1.0 -0.75 -0.5 -0.25 0.0
+        >>> ds.pint.to("g")
+        <xarray.Dataset>
+        Dimensions:  (x: 5)
+        Coordinates:
+            u        (x) int64 [s] 0 1 2 3 4
+        Dimensions without coordinates: x
+        Data variables:
+            a        (x) float64 [g] 0.0 250.0 500.0 750.0 1e+03
+            b        (x) float64 [g] -0.001 -0.00075 -0.0005 -0.00025 0.0
         """
+        if isinstance(units, (str, pint.Unit)):
+            units = {x: units for x in self.ds.keys()}
+            units.update(unit_kwargs)
+            unit_kwargs = None
```

Comment:
I think it's better to be as close to `DataArray.pint.to` as possible:
```suggestion
            unit_kwargs.update(
                {name: units for name in self.ds.keys() if name not in unit_kwargs}
            )
            units = None
```